ui/spec/dashboards/reducers/uiSpec.js

```diff
@@ -50,6 +54,25 @@ describe('DataExplorer.Reducers.UI', () => {
     expect(actual.dashboard).to.deep.equal(d2)
   })
 
+  it('can handle a successful dashboard deletion', () => {
+    const loadedState = reducer(state, loadDashboards(dashboards))
+    const expected = [d1]
+    const actual = reducer(loadedState, deleteDashboard(d2))
+
+    expect(actual.dashboards).to.deep.equal(expected)
+  })
+
+  it('can handle a failed dashboard deletion', () => {
+    const loadedState = reducer(state, loadDashboards([d1]))
+    const actual = reducer(loadedState, deleteDashboardFailed(d2))
+    const actualFirst = _.first(actual.dashboards)
+
+    expect(actual.dashboards).to.have.length(2)
```

Comment:
Agreed on the code smell issue and how we're implementing reducers. Though in this case, they won't be deeply equal because a couple of extra properties are added by the loadDashboards reducers, such as `isEditing`.